From a37eb0a46669592e32ed4e004abb2698ee4f90c5 Mon Sep 17 00:00:00 2001
From: Changqing Li <changqing.li@windriver.com>
Date: Wed, 25 Jul 2018 09:55:25 +0800
Subject: [PATCH] cmocka: fix musl libc conflicting types error

/third_party/cmocka/cmocka.h:126:28: error: conflicting types for 'uintptr_t'
            typedef unsigned int uintptr_t;
                                 ^~~~~~~~~
use __DEFINED_uintptr_t in alltypes.h to check if uintptr already defined

Upstream-Status: Pending

Signed-off-by: Changqing Li <changqing.li@windriver.com>
---
 third_party/cmocka/cmocka.h | 7 +++----
 1 file changed, 3 insertions(+), 4 deletions(-)

diff --git a/third_party/cmocka/cmocka.h b/third_party/cmocka/cmocka.h
index e6861c8..238201d 100644
--- a/third_party/cmocka/cmocka.h
+++ b/third_party/cmocka/cmocka.h
@@ -111,7 +111,7 @@ typedef uintmax_t LargestIntegralType;
     ((LargestIntegralType)(value))
 
 /* Smallest integral type capable of holding a pointer. */
-#if !defined(_UINTPTR_T) && !defined(_UINTPTR_T_DEFINED)
+#if !defined(__DEFINED_uintptr_t)
 # if defined(_WIN32)
     /* WIN32 is an ILP32 platform */
     typedef unsigned int uintptr_t;
@@ -135,9 +135,8 @@ typedef uintmax_t LargestIntegralType;
 #  endif /* __WORDSIZE */
 # endif /* _WIN32 */
 
-# define _UINTPTR_T
-# define _UINTPTR_T_DEFINED
-#endif /* !defined(_UINTPTR_T) || !defined(_UINTPTR_T_DEFINED) */
+# define __DEFINED_uintptr_t
+#endif /* !defined(__DEFINED_uintptr_t) */
 
 /* Perform an unsigned cast to uintptr_t. */
 #define cast_to_pointer_integral_type(value) \
-- 
2.25.1

